### PR TITLE
Add mock errinj unit test

### PR DESCRIPTION
**Pull request by jelonanX**
fpga_reset, 
fpga_port_umsg_enable
fpga_port_umsg_disable 

The ioctl above have not been implemented/turn on as DEFAULT_IOCTL_HANDLER in ioctl_handlers.cpp. Those tests are temporary commented out.

### CODE REVIEW

File: testing/unit/munit-mockerrinj-c.cpp [hunk 1]
```diff
@@ -0,0 +1,277 @@
...
+  uint32_t num_matches_;
+  test_platform platform_;
+  test_system *system_;
+};
+
+
+/**
+* @test    fpga_mock_errinj_01
+* @brief   Tests: fpgaReset
+* @details fpgaReset resets fpga afu
```
Comment (tswhison) [inline testing/unit/munit-mockerrinj-c.cpp]: It's not clear from the wording what this test is trying to accomplish.  What is the error being injected?

File: testing/unit/munit-mockerrinj-c.cpp [hunk 2]
```diff
@@ -0,0 +1,277 @@
...
+TEST_P(mock_err_inj_c_p, fpga_mock_errinj_02) {
+  uint32_t mmio_num;
+  uint64_t *mmio_ptr;
+  
+  // Open port device
+  ASSERT_EQ(FPGA_OK, fpgaOpen(tokens_[0], &handle_, 0));
+  
+  // mmap 
+  mmio_num = 0;
+  EXPECT_EQ(FPGA_OK, fpgaMapMMIO(handle_, mmio_num, &mmio_ptr));
```
Comment (tswhison) [inline testing/unit/munit-mockerrinj-c.cpp]: Not sure what error is being injected?

File: testing/unit/munit-mockerrinj-c.cpp [hunk 3]
```diff
@@ -0,0 +1,277 @@
...
+  uint64_t buf_len;
+  uint64_t* buf_addr;
+  uint64_t wsid = 1;
+  
+  // Open port device
+  ASSERT_EQ(FPGA_OK, fpgaOpen(tokens_[0], &handle_, 0));
+  
+  // Allocate a buffer
+  buf_len = 1024;
+  EXPECT_EQ(FPGA_OK,
```
Comment (tswhison) [inline testing/unit/munit-mockerrinj-c.cpp]: Not sure what error is being injected?

File: testing/unit/munit-mockerrinj-c.cpp [hunk 4]
```diff
@@ -0,0 +1,277 @@
...
+//	EXPECT_NE(FPGA_OK, fpgaTriggerUmsg(h, 0));
+//
+//	// Close port handle
+//	ASSERT_EQ(FPGA_OK, fpgaClose(h));
+//}
+//
+/**
+* @test    fpga_mock_errinj_05
+* @brief   Tests:fpgaAssignPortToInterface
+* @details fpgaAssignPortToInterface Assign and Release port
```
Comment (tswhison) [inline testing/unit/munit-mockerrinj-c.cpp]: Not sure which error is being injected.

**Comment by jelonanX** [comment]
All the tests are testing with uninitialized handle. I will discard them except for the last one which tests fpgaAssignPortToInterface with a valid handle.

**Comment by tswhison** [comment]
Don't discard them, necessarily.  It would also be valid to just update the comments to reflect the scenario under test.

**Pull request closed without merging**